src/main/kotlin/com/dansplugins/factionsystem/claim/MfClaimService.kt

```diff
@@ -60,6 +60,24 @@ class MfClaimService(private val plugin: MedievalFactions, private val repositor
         return false
     }
 
+    // Checks whether a set of chunks has at least one chunk that is adjacent to an existing claim. Works across multiple worlds.
+    fun claimIsAdjacent(chunks: Set<Chunk>, id: MfFactionId): Boolean
+    {
+        for (chunk in chunks)
+        {
+            for (claimedChunk in getClaims(id))
```

Comment:
You don't need to iterate all claims - you can check specifically the chunks around the chunk that you're checking